design-reviews/recipes/task-design-review-template.md

```diff
@@ -0,0 +1,47 @@
+# Your Task Design Title Here (prefix with DRAFT/WIP to indicate level of completeness)
+
+> When developing a design document for a new task, it should contain a detailed design proposal demonstrating how it will solve the goals outlined below.  
+> Not all tasks require a design review, but when they do it is likely that there many unknowns or the solution may be more complex.
+> The design should include diagrams, pseudo code, interface contracts as needed to provide a detailed understanding of the proposal.
+
+* [Task Name](http://link-to-task-work-item)
+* [Story Name](http://link-to-story-work-item)
+* Engagement: [Engagement]
+* Customer: [Customer]
+* Authors: [Author1, Author2, etc.]
+
+## Overview / Problem Statement
+
+* Describe the task with a high-level summary.
+* Consider additional background and justification, for posterity and historical context.
+
+## Goals / In-Scope
+
+* List a few bullet points of what this task will achieve and that are most relevant for the design review discussion.
+* This should include acceptance criteria required to meet the [definition of done](../../team-agreements/definition-of-done/).
+
+## Non-goals / Out-of-Scope
+
+* List a few bullet points of non-goals to clarify the work that is beyond the scope of the design review for this task.
+
+## Proposed Design / Suggested Approach
```

Comment:
I would change the headings to guide the author to fill out a very short description of Options. The reality is if we are doing a design at a task level, we just need to make a decision (not complete a spec and fill out every angle). Since we would typically aim to finish a story in a sprint (1 week), the task needs to be finished in a couple of days. Therefore, I would suggest we focus on people making a decision using summarized information. I would want this to be something the author can finish in a few hours (definitely less than a day). Thoughts @fnocera @arsenvlad ? 

So for example...

## Proposed Options

### Option 1: [`Insert Option Title Here`]
[`TODO: Provide short (2-3 sentances) description how the proposal achieves the goals`]
[`TODO: Provide illustration of the solution (i.e. architecture diagram, flowchart, or sequence). Use minimum needed to convey the suggested change. This should indicate the any changes in the tech stack`]

#### Advantages
[`TODO: Quick bullet list (3 max) what advantages this has over other options`]
#### Risks
[`TODO: Quick bullet list (3 max) what risks if any this has when compared with other options`]

### Option 2: [`Insert Options Title Here`]
[`TODO: complete using same template outlined in option 1`]

Reply:
We don't have a template for user stories, am I correct? I would actually rename this whole contribution to be a template for user stories. In my experience we don't do design for tasks and if we do it is an incremental addition to the design document for the user story or would agree with Charles' suggestion above. 
Thoughts @charleszipp @arsenvlad @manojvazirani

Reply:
@fnocera FYI - The template for using stories is in our backlog and coming soon.

Reply:
@fnocera : #131 is the task for creating template for user stories.

I have updated the template to include proposed options which will be helpful to summarize task level decisions.